base/file.jl

```diff
@@ -416,18 +416,29 @@ function touch(path::AbstractString)
     path
 end
 
-const temp_prefix = "jl_"
-
-if Sys.iswindows()
+"""
+    tempdir()
 
+Obtain the path of a temporary directory (possibly shared with other processes). On Windows,
+`tempdir()` uses the first environment variable found in the ordered list `TMP`, `TEMP`,
+`USERPROFILE`. On all other operating systems, `tempdir()` uses the first environment
+variable found in the ordered list `TMPDIR`, `TMP`, `TEMP`, and `TEMPDIR`. If none of these are
+found, the path `"/tmp"` is used.
+"""
 function tempdir()
-    temppath = Vector{UInt16}(undef, 32767)
-    lentemppath = ccall(:GetTempPathW, stdcall, UInt32, (UInt32, Ptr{UInt16}), length(temppath), temppath)
-    windowserror("GetTempPath", lentemppath >= length(temppath) || lentemppath == 0)
-    resize!(temppath, lentemppath)
-    return transcode(String, temppath)
+    path_max = 1024
+    buf = Vector{UInt8}(undef, path_max)
+    sz = Ref{Csize_t}(path_max)
+    rc = ccall(:uv_os_tmpdir, Cint, (Ptr{UInt8}, Ptr{Csize_t}), buf, sz)
+    @assert rc == 0
```

Comment:
Assertion or systemerror?

Reply:
I'm not 100% sure which is best.  There are two scenarios were `rc!=0`: Invalid inputs to `uv_os_tmpdir`, for which `@assert` is appropriate since this should never happen; and  possible errors  related to buffer having e.g. too small a length, which shouldn't happen, but might, for which throwing an error is probably best?

From the docs

```
 On success size is set to the string length of buffer (which does not include the terminating null). On UV_ENOBUFS failure size is set to the required length for buffer, including the null byte.
```